fix: export Field-related types from Blockly

btw17 opened the pull request:
<!--
  - Thanks for submitting code to Blockly!  Please fill out the following as part of
  - your pull request so we can review your code more easily.
  -->

## The basics

- [x] I branched from develop
- [x] My pull request is against develop
- [x] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)
- [x] I ran `npm run format` and `npm run lint`

## The details
### Resolves

Re-export Field types from Blockly
fixes #6870

### Proposed Changes

- Adds `FieldDropdownConfig` export
- Imports then re-exports `Field{Name}Config` and `Field{Name}FromJsonConfig` types from `./core/blockly.ts`

#### Behavior Before Change

- Config exports were not re-exported in `./core/blockly.ts`

#### Behavior After Change

- Config exports are re-exported in `./core/blockly.ts`

### Reason for Changes

- To make the included types available in the root export for Blockly

### Test Coverage

N/A

### Documentation

N/A

### Additional Information

- `FieldDropdownConfig` is just a re-export of `FieldConfig`. Should we include this?
  - We currently have `FieldTextInputConfig` as a re-export of `FieldInputConfig` already, though the base `FieldInputConfig` isn't actually exported since `FieldInput` is only used internally
- **Note:** As mentioned above, the base `FieldInput` types aren't exported. It is not a registered field since it was created for internal use when splitting the old `FieldInput` to `FieldTextInput`. I don't think these types should be exported, though it's worth noting that this is deliberate.
- In `./core/blockly.ts`, think it's worthwhile to export `UnattachedFieldError` or `FieldProto`?
  - `FieldProto` isn't necessarily needed, though it could theoretically be used by and edge-case subclass that makes significant changes upon the base Field class

maribethb commented:
Changes so far lgtm, regarding the questions (which can be addressed in this PR if you like)

- `FieldInput` and its related objects should all be marked `@internal` if they are intentionally not exported. This removes them from the documentation.
- I think `FieldProto` should also be marked `@internal` and not exported, based on how we're currently using it. If needed someone can file a fr and ask us to change this in the future
- Yes, let's export `UnattachedFieldError`

btw17 commented:
Should be all set @maribethb! I did as you suggested with a few other notes:
- I moved `Validator` types to the bottom of their files so match the convention with the `Config` files
- I removed a redundant comment line for the validator in `core/field.ts`
- I copied the validator function comment to each redeclared type
- I removed the export for `InputTypes` in `./core/field_inputs.ts` since it's not exported in `core/blockly.ts` and not used in any other files

maribethb commented:
Note: Changed the PR title, because I think this should be reflected in release notes (chores are excluded)

Also updated PR comment to close the linked issue.
